list_view/list_view.h

```diff
@@ -58,6 +58,22 @@ class ListView {
         }
     };
 
+    class ItemTransaction {
+    public:
+        ~ItemTransaction();
```

Comment:
```suggestion
        ~ItemTransaction() noexcept;
```
Make destructors noexcept, as they should not fail.

Reply:
Hmm it could throw if one of the `gsl::narrow<>` casts in one of the called functions fails, but that would be a bug (or something is very wrong) if that happens.

As I understand a `noexcept` function will call `std::terminate()` if it does throw (or at least it is supposed), which is fine by me, so I've added it, thanks.

How did you find this repo anyway? The outer class here (ListView) is pretty old and a bit of a mess unfortunately.